### PR TITLE
Userland: Change static const variables to static constexpr

### PR DESCRIPTION
**Pull request by ldm5180**
`static const` variables can be computed and initialized at run-time
during initialization or the first time a function is called. Change
them to `static constexpr` to ensure they are computed at
compile-time.

This allows some removal of `strlen` because the length of the 
`StringView` can be used which is pre-computed at compile-time.

### CODE REVIEW

File: Userland/Demos/Fire/Fire.cpp [hunk 1]
```diff
@@ -45,7 +46,7 @@
 #define FIRE_HEIGHT 200
 #define FIRE_MAX 29
 
-static const Color s_palette[] = {
+static constexpr Array s_palette = {
```
Comment (kleinesfilmroellchen) [inline Userland/Demos/Fire/Fire.cpp:49]: Let's make the type explicit with `Array<Color>`.

File: Userland/Demos/VirGLDemo/CommandBufferBuilder.cpp [hunk 2]
```diff
@@ -235,9 +237,9 @@ void CommandBufferBuilder::append_set_constant_buffer(Vector<float> const& const
     }
 }
 
-void CommandBufferBuilder::append_create_shader(ObjectHandle handle, Gallium::ShaderType shader_type, const char* shader_data)
+void CommandBufferBuilder::append_create_shader(ObjectHandle handle, Gallium::ShaderType shader_type, StringView shader_data)
 {
-    size_t shader_len = strlen(shader_data) + 1; // Need to remember to copy null terminator as well if needed
+    size_t shader_len = shader_data.length() + 1; // Need to remember to copy null terminator as well if needed
```
Comment (kleinesfilmroellchen) [inline Userland/Demos/VirGLDemo/CommandBufferBuilder.cpp:242]: I would like an explanation of why this change is also done in the commit description, as I had to figure it out myself (and it's completely acceptable).

File: Userland/Demos/VirGLDemo/VirGLDemo.cpp [hunk 3]
```diff
@@ -92,7 +95,7 @@ static ResourceID create_virgl_resource(VirGL3DResourceSpec& spec)
 static Vector<VertexData> gen_vertex_data()
 {
     Vector<VertexData> data;
-    static const VertexData vertices[8] = {
+    static constexpr Array vertices = {
```
Comment (kleinesfilmroellchen) [inline Userland/Demos/VirGLDemo/VirGLDemo.cpp:98]: le same deal

File: Userland/Demos/VirGLDemo/VirGLDemo.cpp [hunk 4]
```diff
@@ -102,7 +105,7 @@ static Vector<VertexData> gen_vertex_data()
         VertexData { .r = 0, .g = 0, .b = 0, .x = -0.5, .y = 0.5, .z = 0.5 },
         VertexData { .r = 0, .g = 0, .b = 0, .x = 0.5, .y = 0.5, .z = 0.5 },
     };
-    size_t tris[36] = {
+    static constexpr Array tris = {
```
Comment (kleinesfilmroellchen) [inline Userland/Demos/VirGLDemo/VirGLDemo.cpp:108]: Here, we're probably changing it implicitly to `Array<int>`, which is no bueno.

File: Userland/Libraries/LibAudio/Loader.h [hunk 5]
```diff
@@ -20,8 +20,8 @@
 
 namespace Audio {
 
-static const String empty_string = "";
-static String no_plugin_error = "No loader plugin available";
+static constexpr StringView empty_string = "";
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibAudio/Loader.h:23]: There should be an empty String or StringView available as a static member on one of them.

File: Userland/Libraries/LibGUI/Calendar.cpp [hunk 6]
```diff
@@ -16,21 +17,21 @@ REGISTER_WIDGET(GUI, Calendar);
 
 namespace GUI {
 
-static const char* long_day_names[] = {
+static constexpr Array long_day_names = {
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibGUI/Calendar.cpp:20]: le same deal, all throughout this file

File: Userland/Libraries/LibGUI/FileIconProvider.cpp [hunk 7]
```diff
@@ -178,7 +180,10 @@ Icon FileIconProvider::icon_for_executable(const String& path)
         int image_size;
     };
 
-    static const IconSection icon_sections[] = { { .section_name = "serenity_icon_s", .image_size = 16 }, { .section_name = "serenity_icon_m", .image_size = 32 } };
+    static constexpr Array icon_sections = {
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibGUI/FileIconProvider.cpp:183]: :^)

File: Userland/Libraries/LibGfx/GIFLoader.cpp [hunk 8]
```diff
@@ -16,8 +17,8 @@
 namespace Gfx {
 
 // Row strides and offsets for each interlace pass.
-static const int INTERLACE_ROW_STRIDES[] = { 8, 8, 4, 2 };
-static const int INTERLACE_ROW_OFFSETS[] = { 0, 4, 2, 1 };
+static constexpr Array INTERLACE_ROW_STRIDES = { 8, 8, 4, 2 };
+static constexpr Array INTERLACE_ROW_OFFSETS = { 0, 4, 2, 1 };
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibGfx/GIFLoader.cpp:21]: Force the integer type here as well, just to be sure.

File: Userland/Libraries/LibAudio/Loader.h (hunk 5 above)
Comment (ldm5180) [reply Userland/Libraries/LibAudio/Loader.h]: Turns out, this is unused so I removed it.

**Review by kleinesfilmroellchen — APPROVED** [review]

**Review by kleinesfilmroellchen — APPROVED** [review]
